basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+
+                    current_lanelet = wm->getMapRoutingGraph()->following(current_lanelet, false).front(); 
+                    auto current_lanelet_linestring = current_lanelet.centerline2d().basicLineString();    
+                    reference_centerline.insert(reference_centerline.end(), current_lanelet_linestring.begin(), current_lanelet_linestring.end());
+                    starting_lane.push_back(current_lanelet);
+                }
+            }
+
+            //Find crosstrack distance between starting lane and ending lane
+            std::vector<lanelet::BasicPoint2d> ending_lane_centerline;
+            for(size_t i = 0;i<starting_lane.size();++i){
+                lanelet::ConstLanelet curr_end_lanelet;
+                
+                if(is_lanechange_left){
+                    
+                    //get left lanelet
+                    if(wm->getMapRoutingGraph()->left(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->left(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentLeft(starting_lane[0]).get();
+                    }
+                }
+                else{
+
+                    //get right lanelet
+                    if(wm->getMapRoutingGraph()->right(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->right(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentRight(starting_lane[0]).get();
+                    }
+                }
+                
+                auto ending_lane_linestring = curr_end_lanelet.centerline2d().basicLineString();
+                ending_lane_centerline.insert(ending_lane_centerline.end(), ending_lane_linestring.begin(), ending_lane_linestring.end());
+                
+            }
+
+            cav_msgs::VehicleState ending_state = waypoint_generation::get_nearest_state_to_downtrack(ending_lane_centerline, ending_downtrack, wm);
+
+            int reference_line_start_index = waypoint_generation::get_nearest_point_index(reference_centerline, state);
+            int ending_line_end_index = waypoint_generation::get_nearest_point_index(ending_lane_centerline, ending_state);
+
+            lanelet::BasicLineString2d starting_centerline(reference_centerline.begin() + reference_line_start_index, reference_centerline.end());
+            lanelet::BasicLineString2d ending_centerline(ending_lane_centerline.begin(), ending_lane_centerline.begin() + ending_line_end_index);
+
+            //Convert to vector of BasicPoint2d
+
+                //Downsample centerlines
+            lanelet::BasicLineString2d downsampled_start_centerline;
+            downsampled_start_centerline.reserve(400);
+            downsampled_start_centerline = carma_utils::containers::downsample_vector(starting_centerline, 5);
```

Comment:
Hardcoded downsample should be made a parameter. Additionally, an note/assumption should be placed in the method comments that the user has selected a down sample which gives reasonable distances for spline fitting based on the map in use.